realm/realm-library/src/main/java/io/realm/Realm.java

```diff
@@ -254,16 +254,13 @@ public static void removeDefaultConfiguration() {
     /**
      * Creates a {@link Realm} instance without checking the existence in the {@link RealmCache}.
      *
-     * @param configuration {@link RealmConfiguration} used to create the Realm.
-     * @param globalCacheArray if this is not {@code null} and contains an entry for current schema version,
-     * the {@link BaseRealm#schema#columnIndices} will be initialized with the copy of
-     * the entry. Otherwise, {@link BaseRealm#schema#columnIndices} will be populated
-     * from the Realm file.
+     * @param cache the {@link RealmCache} where to create the realm in.
      * @return a {@link Realm} instance.
      */
-    static Realm createInstance(RealmConfiguration configuration, ColumnIndices[] globalCacheArray) {
+    static Realm createInstance(RealmCache cache) {
+        RealmConfiguration configuration = cache.getConfiguration();
         try {
-            return createAndValidate(configuration, globalCacheArray);
+            return createAndValidate(cache);
```

Comment:
Something about the naming of this method/variable bugs me, but I cannot set the finger on it. I guess at this point it sounds like we are validating the cache, instead of a new Realm instance.

What about `createAndValidateRealmFromCache(cache)` or something like that?